core/lib/parameter_hunter.js

```diff
@@ -13,110 +13,227 @@
-        }
-        else {
-          quotePos += 2;
+        //search for end quote if wrapped in quotes. else search for colon.
+        //everything up to that position will be saved in the keys array.
+        switch (paramString[0]) {
+
+          //need to search for end quote pos in case the quotes wrap a colon
+          case '"':
+          case '\'':
+            wrapper = paramString[0];
+            quotePos = paramString.indexOf(wrapper, 1);
+            break;
+
+          default:
+            colonPos = paramString.indexOf(':');
         }
 
-        //prepare param value for move to paramStringWellFormed var.
-        paramStringTmp = paramString.substring(0, quotePos);
+        if (quotePos > -1) {
+          keys.push(paramString.substring(0, quotePos + 1).trim());
 
-        //unescape any escaped single quotes.
-        paramStringTmp = paramStringTmp.replace(/\\'/g, '\'');
+          //truncate the beginning from paramString and look for a value
+          paramString = paramString.substring(quotePos + 1, paramString.length).trim();
 
-        //escape any double quotes.
-        paramStringTmp = paramStringTmp.replace(/"/g, '\\"');
+          //unset quotePos
+          quotePos = -1;
 
-        //replace the delimiting single quotes with double quotes.
-        paramStringTmp = paramStringTmp.replace(/^'/, '"');
-        paramStringTmp = paramStringTmp.replace(/'$/, '"');
+        } else if (colonPos > -1) {
+          keys.push(paramString.substring(0, colonPos).trim());
 
-        //move param key to paramStringWellFormed var.
-        paramStringWellFormed += paramStringTmp;
-        paramString = paramString.substring(quotePos, paramString.length).trim();
-      }
+          //truncate the beginning from paramString and look for a value
+          paramString = paramString.substring(colonPos, paramString.length);
 
-      //if param value is wrapped in double quotes, just move to paramStringWellFormed var.
-      else if (paramString[0] === '"') {
-        quotePos = paramString.search(/[^\\]"/);
+          //unset colonPos
+          colonPos = -1;
 
-        //except for unclosed quotes to prevent infinite loops.
-        if (quotePos === -1) {
-          quotePos = paramString.length - 1;
+        //if there are no more colons, and we're looking for a key, there is
+        //probably a problem. stop any further processing.
+        } else {
+          paramString = '';
+          break;
         }
-        else {
-          quotePos += 2;
+      }
+
+      //now, search for a value
+      if (paramString[0] === ':') {
+        paramString = paramString.substring([1], paramString.length).trim();
+
+        //since a quote of same type as its wrappers would be escaped, and we
+        //escaped those even further with their unicode, it is safe to look for
+        //wrapper pairs and conclude that their contents are values
+        switch (paramString[0]) {
+          case '"':
+            regex = /^"(.|\s)*?"/;
```

Comment:
This regex can be simplified to `/^"(.)*?"/`. The `.` matches any character, whitespace included.
Same goes for the regex below.
I made this change locally on my machine and re-ran the tests to verify that it does indeed work.